cli/src/commands/extensions/permissions.rs

```diff
@@ -6,13 +6,16 @@ use serde::de::Error as _;
 use serde::{Deserialize, Deserializer, Serialize};
 
 /// Resource permissions.
-#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq)]
-#[serde(transparent)]
-pub struct Permission(Option<Vec<String>>);
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum Permission {
+    List(Vec<String>),
+    Boolean(bool),
```

Comment:
`List` also describes the data type rather than the function of that permission though. So if anything both would have to be changed? What would you suggest as a replacement for that?

I think `Global` is not a good idea because it really doesn't have anything to do with global or local. `AllowAll` is a bit better but I'm not sure if `AllowAll(false)` correctly represents what we want, since that is basically explicitly allowing none, not not allowing all.

Reply:
I agree that `AllowAll(false)` is confusing. So maybe that should be a different variant?

```
pub enum Permission {
    DenyAll,
    Allow(Vec<String>),
    AllowAll,
}
```

Reply:
How would that deserialize without writing a manual deserializer? The advantage of our current solution is that it deserializes with just an annotation.

Reply:
Fair enough... It could be done with `#[serde(from = "SomeType")]`, but it wouldn't be great...